server/helpers/ffmpeg-utils.ts

```diff
@@ -149,3 +188,127 @@ function getVideoFileStream (path: string) {
     })
   })
 }
+
+/**
+ * A slightly customised version of the 'veryfast' x264 preset
+ *
+ * The veryfast preset is right in the sweet spot of performance
+ * and quality. Superfast and ultrafast will give you better
+ * performance, but then quality is noticeably worse.
+ */
+function veryfast (_ffmpeg) {
+  _ffmpeg
+    .preset(standard)
+    .outputOption('-preset:v veryfast')
+    .outputOption(['--aq-mode=2', '--aq-strength=1.3'])
+  /*
+  MAIN reference: https://slhck.info/video/2017/03/01/rate-control.html
+  Our target situation is closer to a livestream than a stream,
+  since we want to reduce as much a possible the encoding burden,
+  altough not to the point of a livestream where there is a hard
+  constraint on the frames per second to be encoded.
+
+  why '--aq-mode=2 --aq-strength=1.3' instead of '-profile:v main'?
+    Make up for most of the loss of grain and macroblocking
+    with less computing power.
+  */
+}
+
+/**
+ * A preset optimised for a stillimage audio video
+ */
+function audio (_ffmpeg) {
+  _ffmpeg
+    .preset(veryfast)
+    .outputOption('-tune stillimage')
+}
+
+/**
+ * A toolbox to play with audio
+ */
+namespace audio {
+  export const get = (_ffmpeg, pos: number | string = 0) => {
```

Comment:
I'm having trouble dealing with the Promise and get stuck with strange timeouts, even if I catch them in my tests:

`Error: Timeout of 60000ms exceeded. For async tests and hooks, ensure "done()" is called; if returning a Promise, ensure it resolves.`

I might need help on that one @Chocobozzz @Jorropo

Reply:
Sorry I'm not very good at typescript/angular (I only got 2 weeks of studing in my school) it's hard for me to understand but I have understand what is namespace but I don't understand how the new Promise at line 234 get res and rej ?
And how the promise is transmitted out side of this function ?